docs/builds.md

```diff
@@ -31,6 +31,45 @@ host OS updates to update the workload containers.  Referring to images
 by tag allows you to use other tooling to dynamically update the workload
 containers.
 
+## Users and groups
+
+This is a highly complex topic.  The short version is that instead of invoking
+e.g. `RUN useradd someuser` in a container build (or indirectly via an RPM
+`%post` script), you should use[sysusers.d](https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#).
+
+(Even better, if this is for code executed as part of a systemd unit, investigate
+ using `DynamicUser=yes`)
+
+However, `sysusers.d` only works for "system" users, not human login users.
+
+There is also [systemd JSON user records](https://systemd.io/USER_RECORD/)
+which can be put into a container image; however at the time of this
+writing while a `sshAuthorizedKeys` field exists, it is not synchronized
+directly in a way that the SSH daemon can consume.
+
+It is likely that at some point in the future the operating system upgrade logic
+(bootc/ostree) will learn to just automatically reconcile changes to `/etc/passwd`.
+
+At the current time, a workaround is to include a systemd unit which automatically
+reconciles things at boot time, via e.g.
+
+```text
+ExecStart=/bin/sh -c 'getent someuser || useradd someuser'
+```
+
+For SSH keys, one approach is to hardcode the SSH authorized keys under `/usr` so it's part of
+the clearly immutable state:
```

Comment:
I'd stress this is something "easily" doable for the root user as the example shows - for "other" users, tying the containerfile to whichever way we actually add user at installation seems a bit off to me (as the users are not there, root is)

Reply:
(I just fear this sets a bad example going forward, and I'd leave ssh keys to the "provisioning/deploy" part

Reply:
Many, if not most deployments will definitely want the flexibility of providing keys externally.  But not all.  And it's also convenient to have the keys part of the container image for those that want to be able to update/rotate it "day 2" without extra tooling.

Reply:
I agree with that, I think it's just weird to add keys to an image for users other than root as there's nothing that says "and then create this user" so these keys are just laying around (no biggie tho, just something I remember also Timothee brought up when I demo'ed the kiosk app)